src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -164,9 +164,25 @@ private bool ShouldSkipCompilation(IMethodNode methodCodeNodeNeedingCode)
 
         public void CompileMethod(IReadyToRunMethodCodeNode methodCodeNodeNeedingCode)
         {
+            bool codeGotPublished = false;
             _methodCodeNode = methodCodeNodeNeedingCode;
 
-            CompileMethodInternal(methodCodeNodeNeedingCode);
+            try
+            {
+                if (!ShouldSkipCompilation(methodCodeNodeNeedingCode))
+                {
+                    CompileMethodInternal(methodCodeNodeNeedingCode);
+                    codeGotPublished = true;
+                }
+            }
+            finally
+            {
+                if (!codeGotPublished)
+                {
+                    PublishEmptyCode();
+                }
+                CompileMethodCleanup();
```

Comment:
Nit: I would have kept `CompileMethodCleanup` in the shared implementation - it's definitely something we don't want to forget to do in any of the consumers (R2R, or full AOT) and having it in the dot file runs the risk of forgetting that.